lib/junk_drawer/rails/bulk_updatable.rb

```diff
@@ -11,14 +11,30 @@ def bulk_update(objects)
       objects = objects.select(&:changed?)
       return unless objects.any?
 
-      changed_attributes = extract_changed_attributes(objects)
-      query = build_query_for(objects, changed_attributes)
+      unique_objects = uniquify_and_merge(objects)
+      changed_attributes = extract_changed_attributes(unique_objects)
+      query = build_query_for(unique_objects, changed_attributes)
       connection.execute(query)
       objects.each(&:clear_changes_information)
     end
 
   private
 
+    def uniquify_and_merge(objects)
+      grouped_objects = objects.group_by(&:id).values
+      grouped_objects.each do |duplicates|
```

Comment:
The name `duplicates` made me think that these were limited to duplicate records we had found, when this actually includes everything, right? In that case I might call it `group` or `objects` instead.

Reply:
renamed to `group`, as `objects` would shadow the method parameter above.